tgui/packages/tgui/interfaces/NtosMessenger/ChatScreen.tsx

```diff
@@ -0,0 +1,407 @@
+import { Stack, Section, Button, Box, Input, Modal } from '../../components';
+import { Component, RefObject, createRef, SFC } from 'inferno';
+import { NtChat, NtMessenger, NtPicture } from './types';
+import { BooleanLike } from 'common/react';
+import { useBackend } from '../../backend';
+
+type ChatScreenProps = {
+  onReturn: () => void;
+  // if one doesn't exist, assume the other one does
+  chat?: NtChat;
+  temp_msgr?: NtMessenger;
+  storedPhotos?: NtPicture[];
+  selectedPhoto: string | null;
+  isSilicon: BooleanLike;
+  sendingVirus: BooleanLike;
+};
+
+type ChatScreenState = {
+  msg: string;
+  selectingPhoto: boolean;
+  canSend: boolean;
+  previewingImage: string | null;
+};
+
+const READ_UNREADS_TIME_MS = 3000;
+const SEND_COOLDOWN_MS = 1000;
+
+export class ChatScreen extends Component<ChatScreenProps, ChatScreenState> {
+  scrollRef: RefObject<HTMLDivElement>;
+  readUnreadsTimeout: NodeJS.Timeout | null;
```

Comment:
Use `?` instead if you can

Reply:
I don't really want to undefine this, assigning null is a bit more meaningful later on